comp/core/tagger/taggerimpl/generic_store/doc.go

```diff
@@ -0,0 +1,15 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package genericstore defines a generic object store that satisfies a redundant use-case in the tagger component implementation.
+// The implementation of the tagger component requires storing objects indexed by keys.
+// Keys are in the form of `{prefix}://{id}`.
+//
+// The package provides a generic interface ObjectStore which can store objects of a given type and index by tagger EntityID (i.e. a prefix + an id).
+// It also provides 2 implementations of this interface:
+// - defaultObjectStore: implements the object store as a plain from entity id to entity object. It is intended to be used when EntityID is stored as a string.
+// - compositeObjectStore: implements the object store as a 2-layered map. The first map is indexed by prefix, and the second map is indexed by id. It is intended to be used when EntityID is stored
+// as a struct separating prefix and id into 2 fields. This implementation is optimised for quick lookups, listing and filtering by prefix.
+package genericstore
```

Comment:
I think you don't have to create doc.go file for adding the package comment 😄 You could add the comment on any file for the generic_store package

Reply:
Thanks 🙇 

I think it is still nice to have the package description in a `doc.go` file. When I read the code again in the future, it will be easier to spot where the detailed description is, instead of checking each file of the package.